GearRent/Networking/APIManager.h

```diff
@@ -6,14 +6,18 @@
 //
 
 #import <Foundation/Foundation.h>
-#import "Item.h"
+#import "Listing.h"
 #import "GNGeoHash.h"
+#import "Category.h"
 
 NS_ASSUME_NONNULL_BEGIN
 
 @interface APIManager : NSObject
 
-void fetchListingsWithCoordinates(NSArray<CLLocation *> *coordinates, void(^completion)(NSArray<Item *> *, NSError *error));
+void fetchListingsWithCoordinates(NSArray<CLLocation *> *coordinates, void(^completion)(NSArray<Listing *> *, NSError *error));
+- (void)fetchNearestCity:(CLLocation *)location completion: (void(^_Nonnull)(NSString *, NSError *)) completion;
```

Comment:
Instead of declaring a instance method, you can declare class method instead so you don't need to alloc APIManager while calling the method. Instead, you can call [APIManager fetchNearestCity: completion:];
```suggestion
+ (void)fetchNearestCity:(CLLocation *)location completion: (void(^_Nonnull)(NSString *, NSError *)) completion;
```

Reply:
When I turn this method into a class method I can't access the fetchNearestCityTask because that is an instance variable.

Reply:
@eninodel  do you need the `fetchNearestCityTask ` instance variable? Can that be a local variable?